service/datastore/api/src/main/java/org/eclipse/kapua/service/datastore/model/query/SortField.java

```diff
@@ -8,43 +8,49 @@
  *
  * Contributors:
  *     Eurotech - initial API and implementation
+ *     Red Hat Inc
  *******************************************************************************/
 package org.eclipse.kapua.service.datastore.model.query;
 
 /**
  * Sortable field definition
- * 
- * @since 1.0
  *
  */
-public interface SortField {
-
-    /**
-     * Get the field name
-     * 
-     * @return
-     */
-    public String getField();
-
-    /**
-     * Set the field name
-     * 
-     * @param field
-     */
-    public void setField(String field);
-
-    /**
-     * Get the sort direction
-     * 
-     * @return
-     */
-    public SortDirection getSortDirection();
-
-    /**
-     * Set the sort direction
-     * 
-     * @param sortDirection
-     */
-    public void setSortDirection(SortDirection sortDirection);
+public class SortField {
 
+    private SortDirection sortDirection;
+    private String field;
+
+    private SortField(final SortDirection direction, final String field) {
+        this.sortDirection = direction;
+        this.field = field;
+    }
+
+    public String getField() {
+        return this.field;
+    }
+
+    public SortDirection getSortDirection() {
+        return this.sortDirection;
+    }
+
+    public static SortField of(SortDirection direction, final String fieldName) {
+        if (fieldName == null || fieldName.isEmpty()) {
+            return null;
```

Comment:
I agree, the UI can omit the sort field. Sorry I missed that this object is exposed outside from the datastore.
I'll check the datastore code to be sure that, if the sort field is null, no sort condition is mapped to ES client layer. If the check is not implemented, I'll open an issue.